Source/BluetoothManager.swift

```diff
@@ -255,8 +255,8 @@ public class BluetoothManager {
     /// - parameter peripheral: The `Peripheral` to which the `BluetoothManager` is either trying to
     /// connect or has already connected.
     /// - returns: `Single` which emits next event when peripheral successfully cancelled connection.
-    public func cancelPeripheralConnection(_ peripheral: Peripheral) -> Single<Peripheral> {
-        let observable = Observable<Peripheral>.create { [weak self] observer in
+    public func cancelPeripheralConnection(_ peripheral: Peripheral) -> Observable<(Peripheral, BluetoothError?)> {
```

Comment:
Maybe we could introduce a wrapper type to indicate that the peripheral might be returned without error?

Reply:
What do you mean by that? Optional type indicates that there may or may not be an error related to disconnection. How would you name your wrapper type? Will it be an enum?

Reply:
Looking through library i found that in most cases a tuple is used instead of wrapper type. I decided to follow the current style. Don't you think that optionality on `BluetoothError` indicates enough that the observable can return without error?

Reply:
in Observable<T>, T is successful object, so for me using tuple there is OK.